### ISSUE TITLE
Unsupported select in hierarchical ref

### ISSUE THREAD

**Issue by alaindargelas** [chipsalliance/synlig#2020]
It looks like we don't support the following: "Selecting a range of positions from a multirange is not supported in the dot notation".
https://github.com/chipsalliance/synlig/blob/bd3586102ffd9374ad299a61e5f71ede6da341a0/frontends/systemverilog/uhdm_ast.cc#L850
Full test with Yosys script:
[illegal_range.tar.gz](https://github.com/chipsalliance/synlig/files/12744324/illegal_range.tar.gz)

Snippet:
```
 always_comb begin : key_sideload_get
    for (int s = 0; s < NumSharesKey; s++) begin
      for (int i = 0; i < NumRegsKey; i++) begin
        key_sideload[s][i] = keymgr_key_i.key[s][i * 32 +: 32];    <<<<<
      end
    end
  end

```

**Comment by alaindargelas** [comment]
Unit test:
```
package keymgr_pkg;
   
 typedef struct packed {
    logic valid;
    logic [1:0][255:0] key;
  } hw_key_req_t;
   
endpackage

module top(input  keymgr_pkg::hw_key_req_t     keymgr_key_i);

   parameter NumRegsKey = 3;
   parameter NumSharesKey = 4;
   
  logic                [NumRegsKey-1:0][31:0] key_sideload [NumSharesKey];

     always_comb begin : key_sideload_get
    for (int s = 0; s < NumSharesKey; s++) begin
      for (int i = 0; i < NumRegsKey; i++) begin
        key_sideload[s][i] = keymgr_key_i.key[s][i * 32 +: 32];
      end
    end
     end

endmodule
```

**Comment by alaindargelas** [comment]
Fixed by #2070
